internal/lib/lib.go

```diff
@@ -274,6 +274,8 @@ func GetNSXTTransportZone() string {
 	return NsxTTzType
 }
 
+var nonDnsLabelRegex = regexp.MustCompile(`[^a-zA-Z0-9_-]+`)
```

Comment:
Please check and confirm if _ is allowed in FQDN or not. Don't think it's DNS compliant.

Reply:
Error message from AviController `labels must start with a letter (a-zA-Z) or a digit and end with a letter or a digit, and label must only have letters, digits and hyphen or underscore as composing characters`